### PR TITLE
Update Recipe Organizer Delete Prompt Text

### LINKED ISSUE #2875: [BUG] - "Are you sure you want to delete this X?" popup text is grammatically incorrect when deleting tags and categories
### First Check

- [X] This is not a feature request.
- [X] I added a very descriptive title to this issue (title field is above this).
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Mealie documentation, with the integrated search.
- [X] I already read the docs and didn't find an answer.
- [X] This issue can be replicated on the demo site (https://demo.mealie.io/).

### What is the issue you are experiencing?

When deleting a tag or category, the text of the confirmation pop-up reads a bit odd. For example, when trying to delete a tag named "vegetarian":

image

I'm asked if I want to delete a vegetarian!

This appears to be due to the "confirm-delete-generic` being used improperly:

https://github.com/mealie-recipes/mealie/blob/342fa6128e0cbd18b1b6b3a44e14514c564e3a1e/frontend/lang/messages/en-US.json#L87

Instead of setting `${name}` to the name of the item, it should instead be set to "tag" or "category". Such that the text correctly reads "Are you sure you want to delete this tag?"

### Steps to Reproduce

1) Create a tag or category with name X.
2) Attempt to delete it.
3) See confirmation pop-up stating "Are you sure you want to delete this X?", which is grammatically incorrect.

### Please provide relevant logs

N/A

### Mealie Version

v1.0.0-RC2 - c9852daf0e213dc4cadf97971c7787cf6fc66dfc

### Deployment

Docker (Linux)

### Additional Deployment Details

_No response_

### PR DESCRIPTION
**Pull request by Kuchenpirat**
## What type of PR is this?

- bug
- cleanup

## What this PR does / why we need it:

As described in #2875 delete prompts might create some funny delete prompts ala "Are you sure you want to delete this vegetarian?" This will change out the text to "Are you sure you want to delete this", move the item to be deleted into its own row and adds the Organizer Type to the Dialog Title. 

<details> 
<summary>Screenshots</summary>
<table>
  <tr>
    <th>Before</th>
    <th>After</th>
  </tr>
  <tr>
    <td></td>
    <td></td>
  </tr>
</table>
</details>

This will also fix an issue where the label delete prompt in the manage data page would put the label in the same row as the delete confirmation text, which was mitigated by wraping it with v-row. This didn't merrit its own pr so it included it in this one. 

## Which issue(s) this PR fixes:

Fixes #2875 

## Testing

Manual testing

### CODE REVIEW

File: frontend/components/Domain/Recipe/RecipeOrganizerPage.vue [hunk 1]
```diff
@@ -5,12 +5,15 @@
...
       v-model="dialogs.delete"
-      :title="$t('general.delete-with-name', { name: deleteTarget.name })"
+      :title="$t('general.delete-with-name', { name: $t(translationKey) })"
       color="error"
       :icon="$globals.icons.alertCircle"
       @confirm="deleteOne()"
     >
-      <v-card-text> {{ $t("general.confirm-delete-generic-with-name", { name: deleteTarget.name }) }} </v-card-text>
+      <v-card-text>
+        {{ $t("general.confirm-delete-generic") }}
```
Comment (ekcom) [inline frontend/components/Domain/Recipe/RecipeOrganizerPage.vue]: Specify what is being deleted in dialog text and better HTML

```suggestion
<p>{{ $t("general.confirm-delete-generic-with-name", { name: $t(translationKey) }) }}</p>
```

**Review by ekcom — APPROVED** [review]
Looks much better :)

Tested locally myself.

File: frontend/components/Domain/Recipe/RecipeOrganizerPage.vue (hunk 1 above)
Comment (boc-the-git) [reply frontend/components/Domain/Recipe/RecipeOrganizerPage.vue]: @Kuchenpirat when you get a chance, please have a look at this :)

Comment (Kuchenpirat) [reply frontend/components/Domain/Recipe/RecipeOrganizerPage.vue]: First of all thanks @ekcom for the feedback. Was bussy during the weekend, so my reply comes in a bit late. 

The only issue i can see with this is that Tag would be capitalized in this case, which is the reason i did not implement it this way. 

So either:
- we say capitalized is ok, and we go foreward with the solution proposed by @ekcom 
- Add a translation string specifially for _confirm-delete-tag_ ... which would be the cleanest
- keep it generic with _confirm-delete-generic_

Looking at it i like the proposed implementation more than leaving it all out, so im plementing it for now, but happy to change to either of the above.

**Comment by boc-the-git** [comment]
From a code perspective I don't see any issue, but do you have updated screenshots of the after?

**Comment by Kuchenpirat** [comment]
updated the PR description

**Review by boc-the-git — APPROVED** [review]
LGTM. Let's not let perfect be the enemy of good.

Nice one @Kuchenpirat, good feedback @ekcom 👍

**Pull request merged**